src/allocator_defrag.h

```diff
@@ -0,0 +1,12 @@
+#ifndef __ALLOCATOR_DEFRAG_H
+#define __ALLOCATOR_DEFRAG_H
+#include "sds.h"
+
+int defrag_jemalloc_init(void);
+void defrag_jemalloc_free(void *ptr, size_t size);
+__attribute__((malloc)) void *defrag_jemalloc_alloc(size_t size);
+unsigned long defrag_jemalloc_get_frag_smallbins(void);
+sds defrag_jemalloc_get_fragmentation_info(sds info);
+void defrag_jemalloc_should_defrag_multi(void **ptrs, unsigned long num);
```

Comment:
Let's leave out "jemalloc" in these names to make this API allocator agnostic. We can conditionally define these functions for the right allocator inside allocator_defrag.c. The caller shouldn't need to worry about which allocator is used.

```suggestion
int defrag_init(void);
void defrag_free(void *ptr, size_t size);
__attribute__((malloc)) void *defrag_alloc(size_t size);
unsigned long defrag_get_frag_smallbins(void);
sds defrag_get_fragmentation_info(sds info);
void defrag_should_defrag_multi(void **ptrs, unsigned long num);
```

Reply:
Also, none of these functions follow our typical naming convention of camel case.

Reply:
will fix